### ISSUE TITLE
Wachter initialization issue

### ISSUE THREAD

**Issue by smachar** [BirkhoffG/jax-relax#46]
The wachter method initializes the first `x_cf` as `cf = jnp.array(x, copy=True)`, this caused me a problem when I tested the method with a synthetic data of two classes (2d normal dist). When the clusters of the two classes do not overlap (Forbenius norm < 10), the method fails to find CFs because the gradients (perturbation) are 0. This due to the loss being 0 from the first iteration and not changing at all as the model makes no error on the `x_cf := x` and the distance between `x` and `x_cf` is also 0.

I propose to add random initialization at first iteration: 
```
key = jax.random.PRNGKey(0)
cf = jnp.array(x, dtype=jnp.float32, copy=True)
cf = cf + jax.random.normal(key, cf.shape) * 0.1
```

**Comment by BirkhoffG** [comment]
Thank you for your feedback @smachar 

The latest `v0.2.9` reflects the changes you suggested.
